algorithms/linfa-clustering/benches/k_means.rs

```diff
@@ -97,7 +97,7 @@ fn k_means_incr_bench(c: &mut Criterion) {
                         .sample_chunks(200)
                         .cycle()
                         .try_fold(None, |current, batch| {
-                            let (model, converged) = clf.fit_with(current, &batch);
+                            let (model, converged) = clf.fit_with(current, &batch).unwrap();
```

Comment:
one idea: you can add a variant `GmmError::NotConverged(model)` and branch it like:
```rust
let model = dataset.
   ....
   .try_fold(None, |current, batch| {
      clf.fit_with(current, &batch)
    });

match model {
    Some(model) | Err(GmmError::NotConverged(model)) => stats.add(model.transform(dataset.records()).mean().unwrap()),
    x => x.unwrap(),
}
```